lib/bot.js

```diff
@@ -208,6 +208,11 @@ class Bot {
         this.ircClient.say(ircChannel, prelude);
       } else if (!message.subtype) {
         text = `<${user.name}> ${text}`;
+      } else if (message.subtype === 'file_share') {
+        text = `<${user.name}> File uploaded ${message.file.permalink} / ${message.file.permalink_public}`;
```

Comment:
permalink_public doesn't seem to work for logged in users before it is "created". On external link creation Slack seems to send file_change event but not file_public event. Additionally file_change has just file id without any other details and slack.rtm.dataStore doesn't seem to have method for looking up file by ID. So this would need some own temporary storage for storing file id + url, then on file_change event poll URL whether it works and finally generate another line to IRC if it succeeds.

So I'd guess showing both links is somewhat reasonable.

I also added tests.